index.d.ts

```diff
@@ -0,0 +1,518 @@
+	// comparable in an overlapping area.
+
+	interface MountOptions {
+		/**
+		 * Set to true to force mount the filesystem (will do an unmount first).
+		 */
+		force?: boolean;
+
+		options?: string[];
+
+		displayFolder?: boolean;
+
+		/**
+		 * Called on filesystem init.
+		 * @param cb 
+		 */
+		init?(cb: (code: number) => void): void;
+
+		/**
+		 * Called before the filesystem accessed a file.
+		 * @param path 
+		 * @param mode 
+		 * @param cb 
+		 */
+		access?(path: string, mode, cb: (code: number) => void): void;
+
+		/**
+		 * Called when the filesystem is being stat'ed.
+		 * @param path 
+		 * @param cb accepts a fs stat object after the return code.
+		 */
+		statfs?(path: string, cb: (code: number, fsStat: FSStat) => void): void;
+
+		/**
+		 * Called when a path is being stat'ed. 
+		 * @param path 
+		 * @param cb accepts a stat object (similar to the one returned in fs.stat
+		 * path, cb)) after the return code.
+		 */
+		getattr?(path: string, cb: (code: number, stats: Stats) => void): void;
+
+		/**
+		 * Called when a file descriptor is being stat'ed. Similar to getattr().
+		 * @param path 
+		 * @param fd 
+		 * @param cb 
+		 */
+		fgetattr?(path: string, fd: number,
+			cb: (code: number, stats: Stats) => void): void;
+
+		/**
+		 * Called when a file descriptor is being flushed.
+		 * @param path 
+		 * @param fd 
+		 * @param cb 
+		 */
+		flush?(path: string, fd: number, cb: (code: number) => void): void;
+
+		/**
+		 * Called when a file descriptor is being fsync'ed.
+		 * @param path 
+		 * @param fd 
+		 * @param datasync 
+		 * @param cb 
+		 */
+		fsync?(path: string, fd: number, datasync,
+			cb: (code: number) => void): void;
+		
+		/**
+		 * Called when a folder descriptor is being fsync'ed. Similar to fsync().
+		 * @param path 
+		 * @param fd 
+		 * @param datasync 
+		 * @param cb 
+		 */
+		fsyncdir?(path: string, fd: number, datasync,
+			cb: (code: number) => void): void;
+
+		/**
+		 * Called when a directory is being listed.
+		 * @param path 
+		 * @param cb accepts an array of file/directory names after the return
+		 * code.
+		 */
+		readdir?(path: string, cb: (code: number, lst: string[]) => void): void;
+
+		/**
+		 * Called when a path is being truncated to a specific size.
+		 * @param path 
+		 * @param size 
+		 * @param cb 
+		 */
+		truncate?(path: string, size: number, cb: (code: number) => void): void;
+
+		/**
+		 * Called when a file via descriptor is being truncated to a specific
+		 * size. Similar to truncate().
+		 * @param path 
+		 * @param fd 
+		 * @param size 
+		 * @param cb 
+		 */
+		ftruncate?(path: string, fd: number, size: number,
+			cb: (code: number) => void): void;
+		
+		/**
+		 * Called when a symlink is being resolved.
+		 * @param path 
+		 * @param cb accepts a pathname (that the link should resolve to) after
+		 * the return code.
+		 */
+		readlink?(path: string,
+			cb: (code: number, targetPathname: string) => void): void;
```

Comment:
In the [usage example](https://www.npmjs.com/package/fuse-bindings#opsreadlinkpath-cb) of `readlink`, it shows cb being used like:
```js
cb(null, 'file.txt')
```

That would have to make the function signature something like:
```ts
cb: (code: number|null) => void
```

Although the usage example may be incorrect and code should actually be `0`, in line with the rest of the methods.

Reply:
@mafintosh this is a question to you.
Does `code` in callbacks take `null` instead of number `0`?
Or, is it the case only with `readlink`?